### PR TITLE
Add new loading animation

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Loading screen

### Description
This PR replaces our loading screen animation with a new custom made.
You can test it by adding a key named "customDelay" with any value to your browser's local storage.

### Steps

- [x] Added implementation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

**Comment by DerMolly** [comment]
Not sure I'm overally happy with the gray color…

**Review by DerMolly — COMMENTED** [review]
Not sure I'm overally happy with the gray color…

**Review by DerMolly — APPROVED** [review]

File: src/components/application-loader/initializers/index.ts [hunk 1]
```diff
@@ -12,7 +12,11 @@ import { fetchFrontendConfig } from './fetch-frontend-config'
 import { loadDarkMode } from './load-dark-mode'
 
 const customDelay: () => Promise<void> = async () => {
-  if (window.localStorage.getItem('customDelay')) {
+  if (
+    typeof window !== 'undefined' &&
+    typeof window.localStorage !== 'undefined' &&
+    (window.location.search.startsWith('?customDelay=') || window.localStorage.getItem('customDelay'))
+  ) {
     return new Promise((resolve) => setTimeout(resolve, 5000))
```
Comment (ErikMichelson) [inline src/components/application-loader/initializers/index.ts:20]: Do we want to use the value of the local storage key / the query parameter as the delay to allow other values than 5 seconds?

Comment (mrdrogdrog) [reply src/components/application-loader/initializers/index.ts:20]: As you can see: no.

File: src/components/application-loader/loading-screen/loading-animation.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        <IconRow></IconRow>
+        <IconRow></IconRow>
+        <IconRow></IconRow>
+        <IconRow></IconRow>
+        <IconRow></IconRow>
+        <IconRow></IconRow>
+        <IconRow></IconRow>
+        <IconRow></IconRow>
+        <IconRow></IconRow>
+        <IconRow></IconRow>
```
Comment (ErikMichelson) [inline src/components/application-loader/loading-screen/loading-animation.tsx:45]: Would it make sense to use `createNumberRangeArray` (and a memo) here?

File: src/components/application-loader/initializers/index.ts (hunk 1 above)
Comment (ErikMichelson) [reply src/components/application-loader/initializers/index.ts:20]: Why not?

Comment (mrdrogdrog) [reply src/components/application-loader/initializers/index.ts:20]: It's only a test flag.

Comment (ErikMichelson) [reply src/components/application-loader/initializers/index.ts:20]: I want to enjoy my freetime watching the animation and not pressing the reload button every 5 seconds...

Comment (ErikMichelson) [reply src/components/application-loader/initializers/index.ts:20]: > It's only a test flag.

Okay then, if you insist on it...

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**